src/streamlink/plugins/bbciplayer.py

```diff
@@ -151,7 +151,10 @@ def login(self, ptrt_url):
         """
 
         def auth_check(res):
-            return ptrt_url in ([h.url for h in res.history] + [res.url])
+            p = urlparse(ptrt_url)
+            return urlunparse(p._replace(scheme="https")) in (
```

Comment:
Since `login` gets called with `self.url` as parameter, it's probably better to update/set `self.url` in the plugin constructor, so that everything else gets updated as well. It's not necessary, but IMO it's cleaner. Up to you though.

Unfortunately, `streamlink.utils.url.update_scheme` hasn't been written with actual scheme updates in mind (as its name would imply), hence your force-push update, but it could be updated with that simple logic, so that plugin authors don't have to use the `ParseResult._replace`. But ignore this here, I just wanted to mention it, as forcing the https scheme is a rather common thing to do (and from what it looks like, some plugins are using it incorrectly).

Reply:
Sounds reasonable to me.

What I think `update_scheme()` does and what it actually does gets me every time!

Another thing I noticed it that this appears to be unnecessary:
```
        if auth_check(session_res):
            log.debug("Already authenticated, skipping authentication")
            return True
```
`login()` is only called once, so I don't see how the plugin could already be authenticated.  I can only think it might be cruft left from previous versions.

Reply:
We should maybe change all scheme-less input URLs to https by default instead of http. Chromium is doing that nowadays too.

Reply:
Yes, it won't be long till all browsers are warning against `http` connections and requiring an exception to continue, so that's probably going to be a good idea at some point.